transformer_engine/pytorch/attention.py

```diff
@@ -6639,6 +6658,11 @@ def forward(
                 for x in (query_layer, key_layer, value_layer)
             )
 
+            if self.qkv_format == "thd":
+                query_layer, key_layer, value_layer = (
+                    x.reshape(x.size(0), x.size(2), x.size(3))
+                    for x in (query_layer, key_layer, value_layer)
+                )
```

Comment:
Seems strange to first always do the sbhd/bshd logic and then trying overwriting it with thd logic. Could we do
```python
            if self.qkv_format == "thd":
                query_layer, key_layer, value_layer = (
                    x.reshape(x.size(0), x.size(2), x.size(3))
                    for x in (query_layer, key_layer, value_layer)
                )
            else:
                query_layer, key_layer, value_layer = (
                    x.reshape(x.size(0), x.size(1), -1, self.hidden_size_per_attention_head)
                    for x in (query_layer, key_layer, value_layer)
                )
```
instead?